src/routes/settings/components/Settings.js

```diff
@@ -0,0 +1,197 @@
+  title: string, description: string,
+  hasSwitch?: true,
+  onPress?: () => void
+}
+
+type SectionType = { title: ?string }
+
+const ItemSeparator = styled.View`
+    background-color: ${props => props.theme.colors.textDecorationColor};;
+    height: ${StyleSheet.hairlineWidth};
+`
+
+const SectionHeaderContainer = styled.View`
+`
+
+const SectionHeader = styled.Text`
+    padding: 20px;
+    color: ${props => props.theme.colors.textColor};
+`
+
+export default class Settings extends React.Component<PropsType, StateType> {
+  sections = () => {
+    const {t, language} = this.props
+
+    return ([
+      {
+        title: 'Placeholder',
+        data: [
+          {
+            title: 'Placeholder',
+            description: 'Placeholder'
+          },
+          {
+            title: 'Placeholder',
+            description: 'Placeholder'
+          }
+        ]
+      },
+      {
+        data: [
+          {
+            title: t('troubleshooting'),
+            description: t('troubleshootingDescription'),
+            hasSwitch: true,
+            onPress: () => { this.toggleErrorTracking() }
+          },
+          {
+            title: t('about'),
+            onPress: () => {
+              if (language === 'de') {
+                Linking.openURL('https://integreat-app.de/')
+              } else {
+                Linking.openURL('https://integreat-app.de/en/')
+              }
+            }
+          },
+          {
+            title: t('privacyPolicy'),
+            onPress: () => {
+              if (language === 'de') {
+                Linking.openURL('https://integreat-app.de/datenschutz-webseite/')
+              } else {
+                Linking.openURL('https://integreat-app.de/en/privacy-website/')
+              }
+            }
+          },
+          {
+            title: t('version', {version: '??'})
+          },
+          {
+            title: t('openSourceLicenses'),
+            onPress: () => { console.warn('Not yet implemented.') }
+          }
+        ]
+      }
+    ])
+  }
+
+  constructor (props: PropsType) {
+    super(props)
+
+    this.state = {settingsLoaded: false, settings: {errorTracking: false, test: false}}
+
+    this.loadSettings()
+  }
+
+  async loadSettings () {
+    try {
+      const settingsKeys = Object.keys(this.state.settings)
+      const settingsArray = await AsyncStorage.multiGet(settingsKeys)
+
+      const settings = reduce(settingsArray,
+        (accumulator, [key, value]) => {
+          accumulator[key] = JSON.parse(value)
+          return accumulator
+        },
+        {})
+
+      this.setState(state => ({...state, settingsLoaded: true, settings}))
+    } catch (e) {
+      console.error('Failed to load settings.')
+    }
+  }
+
+  async setSetting (changeSetting: SettingsType => $Shape<SettingsType>) {
+    this.setState(state => {
+      const newSettings = changeSetting(state.settings)
+      return ({...state, settings: {...state.settings, ...newSettings}})
```

Comment:
Could be simplified to `this.setState({settings: {...changeSetting(state.settings)}})`

Reply:
Not sure about this one... I think React only shallowly merges the state. So you could simplify it to
```js
this.setState(state => {settings: {...state.settings, ...changeSetting(state.settings)}})
```

Reply:
Both do not work because they would overwrite `settingsLoaded`

You can only set the whole state with `setState`

Reply:
Have you tried? Usually setState merges shallowly!